cmd/build_sensorbee/main_test.go

```diff
@@ -0,0 +1,180 @@
+					SubCommands: map[string]commandDetail{
+						"run":      commandDetail{},
+						"shell":    commandDetail{},
+						"topology": commandDetail{},
+						"exp":      commandDetail{},
+						"runfile":  commandDetail{},
+					},
+					Version: version.Version,
+				}
+				So(*conf, ShouldResemble, expectedConf)
+			})
+		})
+	})
+}
+
+func TestCreateMainFile(t *testing.T) {
+	dir, err := ioutil.TempDir("", "build_sensorbee_create_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	outFilename := filepath.Join(dir, "test_main.go")
+
+	set := flag.NewFlagSet("dummy", flag.ExitOnError)
+	set.String("source-filename", outFilename, "")
+	c := cli.NewContext(cli.NewApp(), set, nil)
+
+	Convey("Given build_sensorbee tool", t, func() {
+		Convey("When create a main file with a plugin and a buildin command", func() {
+			config := &Config{
+				PluginPaths: []string{"path/to/plugin"},
+				SubCommands: map[string]commandDetail{
+					"run": commandDetail{},
+				},
+				Version: version.Version,
+			}
+			So(create(c, config), ShouldBeNil)
+			Convey("Then the main file should be created", func() {
+				b, err := ioutil.ReadFile(outFilename)
+				So(err, ShouldBeNil)
+				expectedMainFile := `package main
+
+import (
+	_ "gopkg.in/sensorbee/sensorbee.v0/bql/udf/builtin"
+	"gopkg.in/sensorbee/sensorbee.v0/cmd/lib/run"
+	"gopkg.in/sensorbee/sensorbee.v0/version"
+	"gopkg.in/urfave/cli.v1"
+	"os"
+	_ "path/to/plugin"
+	"time"
+)
+
+func init() {
+	// TODO
+	time.Local = time.UTC
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "sensorbee"
+	app.Usage = "SensorBee built with build_sensorbee ` + version.Version + `"
+	app.Version = version.Version
+	app.Commands = []cli.Command{
+		run.SetUp(),
+	}
+	if err := app.Run(os.Args); err != nil {
+		os.Exit(1)
+	}
+}
+`
+				So(string(b), ShouldEqual, expectedMainFile)
+			})
+		})
+
+		Convey("When create a main file with a plugin and a custom command", func() {
+			config := &Config{
+				PluginPaths: []string{"path/to/plugin"},
+				SubCommands: map[string]commandDetail{
+					"repo1": commandDetail{Path: "path/to/repo"},
+				},
+				Version: version.Version,
+			}
+			So(create(c, config), ShouldBeNil)
+			Convey("Then the main file should be created", func() {
+				b, err := ioutil.ReadFile(outFilename)
+				So(err, ShouldBeNil)
+				expectedMainFile := `package main
+
+import (
+	_ "gopkg.in/sensorbee/sensorbee.v0/bql/udf/builtin"
+	"gopkg.in/sensorbee/sensorbee.v0/version"
+	"gopkg.in/urfave/cli.v1"
+	"os"
+	_ "path/to/plugin"
+	repo1 "path/to/repo"
+	"time"
+)
+
+func init() {
+	// TODO
+	time.Local = time.UTC
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "sensorbee"
+	app.Usage = "SensorBee built with build_sensorbee ` + version.Version + `"
+	app.Version = version.Version
+	app.Commands = []cli.Command{
+		repo1.SetUp(),
+	}
+	if err := app.Run(os.Args); err != nil {
+		os.Exit(1)
+	}
+}
+`
+				So(string(b), ShouldEqual, expectedMainFile)
+			})
+		})
+	})
```

Comment:
How about adding a test using multiple `SubCommands`? It doesn't have to perform an exact match of the contents but only needs to look up substrings like `repo1 "path/to/repo"` (possibly with regexp) to see if all subcommands are correctly imported.